network-config-analyzer/IstioGenericYamlParser.py

```diff
@@ -0,0 +1,101 @@
+#
+# Copyright 2020- IBM Inc. All rights reserved
+# SPDX-License-Identifier: Apache2.0
+#
+
+from GenericYamlParser import GenericYamlParser
+from PeerContainer import PeerContainer
+
+
+class IstioGenericYamlParser(GenericYamlParser):
+    """
+    A class for istio yaml parser , common methods for istio policies parsers
+    """
+
+    # TODO: istio_root_namespace should be configurable from istio configuration, currently using default value for it
+    # If namespace is set to istio root namespace, the policy object applies to all namespaces in a mesh
+    istio_root_namespace = 'istio-config'
+
+    def __init__(self, policy, peer_container, file_name=''):
+        """
+        :param dict policy: The istio policy object as provided by the yaml parser
+        :param PeerContainer peer_container: The policy will be evaluated against this set of peers
+        :param str file_name: The name of the file in which the istio policy object resides
+        """
+        GenericYamlParser.__init__(self, file_name)
+        self.policy = policy
+        self.peer_container = peer_container
+        self.namespace = None
+        self.referenced_labels = set()
+
+    def parse_generic_istio_policy_fields(self, policy_kind, istio_version):
+        """
+        Parse the common fields in istio policies, e.g kind, apiVersion and metadata
+        :param str policy_kind : the kind of current policy
+        :param str istio_version : the apiVersion of the istio object
+        :return: the name of the current object or None if it is not relevant object
+        :rtype: str
+        """
+        if not isinstance(self.policy, dict):
+            self.syntax_error('type of Top ds is not a map')
+        if self.policy.get('kind') != policy_kind:
+            return None  # Not the relevant object
+        api_version = self.policy.get('apiVersion')
+        if 'istio' not in api_version:
+            return None  # apiVersion is not properly set
+        valid_keys = {'kind': [1, str], 'apiVersion': [1, str], 'metadata': [1, dict], 'spec': [0, dict]}
+        self.check_fields_validity(self.policy, policy_kind, valid_keys,
+                                   {'apiVersion': [istio_version]})
+        metadata = self.policy['metadata']
+        self.check_metadata_validity(metadata)
+        ns_name = metadata.get('namespace', 'default')
+        warn_if_missing = ns_name != self.istio_root_namespace
+        self.namespace = self.peer_container.get_namespace(ns_name, warn_if_missing)
```

Comment:
We may have some issue when changing the topology (peer container) based on sidecar with root namespace.
If some network config has such a sidecar resource, and we run forbids/permits query with a second network config with similar topology that does not have such sidecar resource, it may be not comparable with different topologies.

Reply:
would excluding the istio_root_namespace from the `__eq__` def in the `PeerContainer` help?
or maybe in order to make it general , I might save the namespaces which doesn't need to be warn about in `get_namespace`  into a `special_namespaces` dict and then in `__eq__` to exclude them from the "regular" namespaces comparison

Reply:
Let's add a test to demonstrate this problem, and create a separate issue to resolve this later, in a different branch.
(Maybe we can handle conflicting topologies when only certain namespaces are missing, in relevant queries that currently do not support that).

Reply:
committed a new scheme file with a "not executed" query

Reply:
great, thanks. 
please also create a new issue with some description of this problem + reference to this test example (which should be modified after the issue is resolved).